tensorflow/python/ops/array_ops.py

```diff
@@ -1084,6 +1084,7 @@ def boolean_mask(tensor, mask, name="boolean_mask"):
   Args:
     tensor:  N-D tensor.
     mask:  K-D boolean tensor, K <= N and K must be known statically.
+    axis: The axis in `tensor` to mask from. Defaults to the first dimension.
```

Comment:
@josh11b If mask isn't 1-D then it should work as well. I added an additional test case for this.